go/libraries/doltcore/doltdb/commit.go

```diff
@@ -196,30 +256,37 @@ func (c *Commit) CanFastReverseTo(ctx context.Context, new *Commit) (bool, error
 	return false, nil
 }
 
-func (c *Commit) GetAncestor(ctx context.Context, as *AncestorSpec) (*Commit, error) {
+func (c *Commit) GetAncestor(ctx context.Context, as *AncestorSpec) (*OptionalCommit, error) {
+	addr, err := c.HashOf()
+	if err != nil {
+		return nil, err
+	}
+	optInst := &OptionalCommit{c, addr}
 	if as == nil || len(as.Instructions) == 0 {
-		return c, nil
+		return optInst, nil
 	}
 
-	cur := c
-
+	hardInst := c
 	instructions := as.Instructions
 	for _, inst := range instructions {
-		if inst >= cur.NumParents() {
+		if inst >= hardInst.NumParents() {
 			return nil, ErrInvalidAncestorSpec
 		}
 
 		var err error
-		cur, err = cur.GetParent(ctx, inst)
+		optInst, err = hardInst.GetParent(ctx, inst)
 		if err != nil {
 			return nil, err
 		}
-		if cur == nil {
-			return nil, ErrInvalidAncestorSpec
+
+		var ok bool
+		hardInst, ok = optInst.ToCommit()
+		if !ok {
+			break
 		}
 	}
 
-	return cur, nil
+	return optInst, nil
```

Comment:
yeah, I thought about this, and I think this is the right way to do it - so long as the callers all handle the error in the way that makes sense for them. In this case, Resolve calls getAncestor, and lots of things lead to Resolve.

Also, I don't really know how we'd handle this otherwise. I was thinking that instead of including the hash in the OptionalCommit structure, we'd actually have the partial ancestor spec. So if you requested HEAD~5, and you could only get to HEAD~2, then we'd return hash_of_ghost~3, but that seems like it wouldn't be very useful to any caller.